chain/sync.go

```diff
@@ -845,6 +845,14 @@ loop:
 	if err != nil {
 		return nil, xerrors.Errorf("failed to load next local tipset: %w", err)
 	}
+
+	if chkpt := syncer.store.GetCheckpoint(); !ignoreCheckpoint && chkpt != nil {
+		// don't allow us to expand the tipset beyond the checkpoint
+		if known.Equals(chkpt) && base.IsChildOf(knownParent) {
+			return nil, xerrors.Errorf("tispet expanding fork: %w", ErrForkCheckpoint)
+		}
+	}
```

Comment:
```suggestion
	if !ignoreCheckpoint {
		if chkpt := syncer.store.GetCheckpoint(); chkpt != nil && base.Height() <= chkpt.Height() {
			// don't allow us to expand the tipset beyond the checkpoint
			return nil, xerrors.Errorf("tispet at or below checkpoint height: %w", ErrForkCheckpoint)
		}
	}
```

Reply:
This wasn't as trivial, needed to make one more change.